src/program.py

```diff
@@ -30,200 +30,164 @@
 
 class Program:
     def __init__(self, repo, name=None, installed=False):
-        '''
-        @param name (String) Name of the program
-        @param installed (bool) If the program is installed or not
-        '''
-        
-        ##Type checking params
+        ## Constructor
+        # @param name (String) Name of the program
+        # @param installed (bool) If the program is installed or not
+
+        ## Type checking params
         if not isinstance(repo, AbsRepoType):
             raise TypeError(u"Program 'repo' expects a repo type e.g. GitType")
-        
+
         # TODO name could be obmitted and recovered from directory/repo?
         if not isinstance(name,(str,unicode)):
             raise TypeError(u"Program 'name' expects a string")
 
         if not isinstance(installed,bool):
-            raise TypeError(u"Program 'installed' expects a bool")       
+            raise TypeError(u"Program 'installed' expects a bool")
 
         self.repo_ref=repo
         self.name=unify(name)
         self.directory=directory
         self.installed=installed
-    
-    
-    
+
     def getCurrentHead(self):
-        '''
-        Returns the latest version or commit for the software
-        '''
-        
+        ## Returns the latest version or commit for the software
+
         return self.repo_ref.getHead()
-    
+
     def getName(self):
-        '''
-        Gets the name of the repo/program
-        '''
-        
+        ## Gets the name of the repo/program
+
         return self.name
-    
-    
-    
-    
+
     ##
     # TODO - Includes renaming/re-factoring into current code
     ##
-    
-    def getVersionFromChangelog(self):
-        '''
-        * @brief Gets a list of versions from a changelog
-        * @param changelog_filename Filename of the changelog
-        * @param prog Program object
-        * @returns zero on success
-        */
-        int program_get_versions_from_changelog()
-        '''
+
+    def FromChangelog(self):
```

Comment:
Not sure on the official name of this style, but most languages discourage capitalising the first letter for functions/variables. Objects start with a capital